impress-org/givewp
# Fix: load translations correctly

## Description



This PR changed how the translation files are loaded because it was using the wrong path in some cases. Using the Loco plugin to test different places is possible to confirm that now the translations are working in the following places: 

WP_LANG_DIR - Global file: `languages/plugins/give-pt_BR.po`

WP_PLUGIN_DIR - Local file: `plugins/give/languages/give-pt_BR.po`

Custom DIR: `languages/loco/plugins/give-pt_BR.po`

In any of these places, the translations are working and the form getting translated as you can see on the attached screenshots a version of the donation form and confirmation page in Portuguese.

To make the translation proper work on JS files that contain strings to be translated, was necessary to use the  `Language::setScriptTranslations($handle)` method to set the proper translation file to the scripts.

## Affects



The `load_textdomain()` function and the scripts related to the new v3 forms.

## Visuals



![donationForm-pt_BR](https://github.com/impress-org/givewp/assets/16308864/61a47ac8-cbcb-460f-b367-33a2ac8f8c3a)

![confirmationPage-pt_BR](https://github.com/impress-org/givewp/assets/16308864/f733b17e-770e-4dce-a547-2d6cd6469f19)

![image](https://github.com/impress-org/givewp/assets/16308864/4c78b254-86cd-43b7-aec6-8f60f96de2f3)


## Testing Instructions



1. Compile the assets using the `npm run build` command;
2. Create the pot file with the `wp i18n make-pot . languages/give.pot` command;
3. Use the Loco translate to change a few strings from the V3 forms;
4. Use Loco Translate to move the translation files to different places;
5. The translated strings should always be displayed.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205303057234928
  - https://app.asana.com/0/0/1204798091678757
